apps/api/src/app/topics/use-cases/rename-topic/rename-topic.use-case.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { TopicEntity, TopicRepository } from '@novu/dal';
+
+import { RenameTopicCommand } from './rename-topic.command';
+
+import { GetTopicUseCase, GetTopicCommand } from '../get-topic';
+import { TopicDto } from '../../dtos/topic.dto';
+
+@Injectable()
+export class RenameTopicUseCase {
+  constructor(private getTopicUseCase: GetTopicUseCase, private topicRepository: TopicRepository) {}
+
+  async execute(command: RenameTopicCommand): Promise<TopicDto> {
+    await this.getTopicUseCase.execute(command);
+
+    const query = this.mapToQuery(command);
+    const renamedTopic = await this.topicRepository.renameTopic(
+      query._id,
+      query._environmentId,
+      query._organizationId,
+      query._userId,
```

Comment:
A small note regarding `_userId`: it's un-necessary and can be entirely removed from schema. It can also introduce  some bugs when different API key's are generated for different users.

Reply:
the topic entity doesn't even have the `_userId` field so it looks like a typo here